File: content/03/code/ruby/01-add-basic-working-scenario-from-last-chapter/features/step_definitions/steps.rb
require 'shouty'

Given("Lucy is located {int}m from Sean") do |distance|
  @lucy = Shouty::Person.new
  @sean = Shouty::Person.new
  @lucy.move_to(distance)
end

When("Sean shouts {string}") do |message|
  @sean.shout(message)
  @message_from_sean = message
end

Then("Lucy hears Sean's message") do
  expect(@lucy.messages_heard).to eq [@message_from_sean]
end

File: content/02/code/js/12-add-second-scenario-see-it-fail/features/step_definitions/steps.js
const { assertThat, is } = require('hamjest')
const Person = require('../../src/shouty.js')
const { Given, When, Then } = require('@cucumber/cucumber')

Given('Lucy is located {int} metres from Sean', function (distance) {
  this.lucy = new Person
  this.sean = new Person
  this.lucy.moveTo(distance)
})

When('Sean shouts {string}', function (message) {
  this.sean.shout(message)
  this.message = message
})

Then('Lucy hears Sean\'s message', function () {
  assertThat(this.lucy.messagesHeard(), is([this.message]))
})


